File: programacao/python/00-conceitos/00_hello/hello_world.py
"""
Programa Hello World - Primeiro programa em Python

Este é o programa mais simples que podemos escrever em Python.
Ele demonstra a estrutura básica de uma aplicação Python.

@author luiscaparroz
@version 1.0
"""

def main():
    """
    Função principal - Ponto de entrada da aplicação
    
    Esta função é chamada quando executamos o programa.
    Em Python, não é obrigatório ter uma função main, mas é uma boa prática.
    """
    # print() imprime uma mensagem no console
    # Por padrão, print() adiciona uma quebra de linha no final
    print("Hello, world")
    
    # Demonstração: podemos imprimir múltiplas linhas
    print("Bem-vindo ao mundo da programação Python!")
    print("Este é meu primeiro programa.")
    
    # Por padrão, print() sempre quebra linha
    # Para não quebrar, use o parâmetro end
    print("Esta mensagem não quebra linha... ", end="")
    print("Esta continua na mesma linha!")
    print()  # Quebra de linha vazia
    
    # Python também pode receber argumentos da linha de comando
    import sys
    if len(sys.argv) > 1:
        print("Você passou os seguintes argumentos:")
        for i, arg in enumerate(sys.argv[1:], 1):
            print(f"Argumento {i}: {arg}")
    else:
        print("Nenhum argumento foi passado pela linha de comando.")
        print("Tente executar: python3 hello_world.py seu_nome")


# Verifica se o script está sendo executado diretamente
# Esta é a forma pythônica de definir o ponto de entrada
if __name__ == "__main__":
    main()

File: programacao/java/00-conceitos/hello_world/src/HelloWorld.java
/**
 * Programa Hello World - Primeiro programa em Java
 * 
 * Este é o programa mais simples que podemos escrever em Java.
 * Ele demonstra a estrutura básica de uma aplicação Java.
 * 
 * @author luiscaparroz
 * @version 1.0
 * @since JDK 1.8
 */
public class HelloWorld {

    /**
     * Método principal (main) - Ponto de entrada da aplicação
     * 
     * Este método é chamado automaticamente quando executamos o programa.
     * Todas as aplicações Java devem ter um método main com esta assinatura exata.
     * 
     * @param args - Argumentos passados pela linha de comando (array de strings)
     */
    public static void main(String[] args) {
        // System.out.println() imprime uma mensagem no console
        // System = classe que representa o sistema operacional
        // out = saída padrão (console/terminal)
        // println = método que imprime uma linha com quebra de linha no final
        System.out.println("Hello, world");
        
        // Demonstração: podemos imprimir múltiplas linhas
        System.out.println("Bem-vindo ao mundo da programação Java!");
        System.out.println("Este é meu primeiro programa.");
        
        // Diferença entre print() e println()
        System.out.print("Esta mensagem não quebra linha... ");
        System.out.print("Esta continua na mesma linha!");
        System.out.println(); // Quebra de linha vazia
        
        // Exemplo usando argumentos da linha de comando
        if (args.length > 0) {
            System.out.println("Você passou os seguintes argumentos:");
            for (int i = 0; i < args.length; i++) {
                System.out.println("Argumento " + (i + 1) + ": " + args[i]);
            }
        } else {
            System.out.println("Nenhum argumento foi passado pela linha de comando.");
            System.out.println("Tente executar: java HelloWorld seu_nome");
        }
    }
}
